### DISCUSSION TITLE
Safety of redis sharing across multiple ArgoCD instances

### DISCUSSION THREAD

**Discussion by msobkowiak-olx** [argoproj/argo-cd#9483] [General]
Hey !

We are currently exploring multi-cluster and multi-argo architecture in our company. To save a bunch on the cache setup (since we intend to later on spin up more ArgoCD instances for project isolation) do you know how safe it is to actually reuse the same redis instance (or cluster) with multiple ArgoCD instances ? Are key collisions a possible problem with such setup ?

Another question to this may be - can ArgoCD select Redis database so key collision is not a problem ?

**Comment by crenshaw-dev** [comment]
Key collision is certainly a potential problem. And since Argo CD doesn't encrypt the data at rest in Redis, that's a security problem.

I'd be interested to see a proposal/PR/enhancement request for key prefixing/value encryption!
